### DISCUSSION TITLE
help request: Failed installing dependency lua-typeof

### DISCUSSION THREAD

**Discussion by zlhgo** [apache/apisix#8225] [Q&A] [answered]
### Description

Installing https://luarocks.org/lua-typeof-0.1-0.rockspec
正克隆到 'lua-typeof'...
fatal: 无法连接到 github.com：
github.com[0: 20.205.243.166]: errno=连接超时

Error: Failed installing dependency: https://luarocks.org/lua-resty-etcd-1.8.3-0.src.rock - Failed installing dependency: https://luarocks.org/lua-typeof-0.1-0.rockspec - Failed cloning git repository.
make: *** [Makefile:153：deps] 错误 1

https://luarocks.org/luaxxhash-1.0.0-1.rockspec
```
package = "lua-typeof"
version = "0.1-0"
source = {
   url = "git://github.com/iresty/lua-typeof",  # 或许应该是  git+https://github.com/iresty/lua-typeof  ?
   tag = "v0.1"
}
description = {
   summary = "Check the data type for Lua variable",
   homepage = "https://github.com/iresty/lua-typeof",
   license = "Apache License 2.0",
   maintainer = "Yuansheng Wang <membphis@gmail.com>"
}
build = {
   type = "builtin",
   modules = {
      ["typeof"] = "lib/typeof.lua",
   }
}
```

### Environment

- APISIX version (run `apisix version`): 2.13.3
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`): 3.9.1

**Comment by tzssangglass** [comment, marked as answer]
ref: https://github.com/apache/apisix/blob/master/docs/zh/latest/FAQ.md#%E4%BD%BF%E7%94%A8-luarocks-%E5%AE%89%E8%A3%85-apache-apisix-%E4%BE%9D%E8%B5%96%E9%A1%B9%E6%97%B6%E4%B8%BA%E4%BB%80%E4%B9%88%E4%BC%9A%E5%AF%BC%E8%87%B4%E8%B6%85%E6%97%B6%E5%AE%89%E8%A3%85%E7%BC%93%E6%85%A2%E6%88%96%E5%AE%89%E8%A3%85%E5%A4%B1%E8%B4%A5

**Reply by zlhgo** [reply]
$ make deps ENV_LUAROCKS_SERVER=https://luarocks.cn

Installing https://luarocks.cn/lua-typeof-0.1-0.rockspec
正克隆到 'lua-typeof'...
fatal: 无法连接到 github.com：
github.com[0: 20.205.243.166]: errno=连接超时


Error: Failed installing dependency: https://luarocks.cn/lua-resty-etcd-1.8.3-0.src.rock - Failed installing dependency: https://luarocks.cn/lua-typeof-0.1-0.rockspec - Failed cloning git repository.
make: *** [Makefile:153：deps] 错误 1

还是一样的。

**Reply by tzssangglass** [reply]
That's a problem with your network environment, try to solve the network problem first.

**Reply by zlhgo** [reply]
```
git config --global url.https://github.com/.insteadOf git://github.com/
```
Solved this problem.

**Reply by tzssangglass** [reply]
maybe you are not use https://github.com/apache/apisix/blob/master/utils/linux-install-luarocks.sh

**Comment by weshho** [comment]
lua-resty-etcd 1.8.2-0 depends on lua-typeof 0.1 (not installed)
Installing https://luarocks.cn/lua-typeof-0.1-0.rockspec
正克隆到 'lua-typeof'...
fatal: unable to access 'https://github.com/iresty/lua-typeof/': Failed connect to github.com:443; Connection timed out

Error: Failed installing dependency: https://luarocks.cn/lua-resty-etcd-1.8.2-0.src.rock - Failed installing dependency: https://luarocks.cn/lua-typeof-0.1-0.rockspec - Failed cloning git repository.
make: *** [deps] 错误 1

我也遇到相同的问题

**Reply by zlhgo** [reply]
那你也用相同的方法解决呗。

**Reply by weshho** [reply]
git config --global url.https://github.com/.insteadOf git://github.com/
可以了，刚才试了没成功，这会又行了
